src/algebra/homology/local_cohomology.lean

```diff
@@ -186,4 +199,67 @@ begin
        ... ≤ J           : hk,
 end
 
+/-- The diagram of powers of `J` is initial in the diagram of all ideals with
+radical containing `J`. This uses noetherianness. -/
+/- Can this proof be golfed? -/
+instance ideal_powers_initial [hR : is_noetherian R R] :
+  functor.initial (ideal_powers_to_self_le_radical J) :=
+{ out := λ J', begin
+  apply @zigzag_is_connected _ _ _,
+  { intros j1 j2,
+    apply relation.refl_trans_gen.single,
+    -- The inclusions `J^n1 ≤ J'` and `J^n2 ≤ J'` always form a triangle, based on
+    -- which exponent is larger.
+    cases le_total (unop j1.left) (unop j2.left) with h,
+      right, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩,
+      left, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩ },
+  { obtain ⟨k, hk⟩ := ideal.exists_pow_le_of_le_radical_of_fg J'.2
+      (is_noetherian_def.mp hR _),
+    refine ⟨costructured_arrow.mk _⟩,
+    exact (op k),
+    exact ⟨⟨hk⟩⟩ },
+  end }
+
+/-- Local cohomology (defined in terms of powers of `J`) agrees with local
+cohomology computed over all ideals with radical containing `J`. -/
+def iso_self_le_radical (J : ideal R) [is_noetherian R R] (i : ℕ) :
+  local_cohomology.of_self_le_radical J i ≅ local_cohomology J i :=
+(local_cohomology.iso_of_final.{u u 0}
+  (ideal_powers_to_self_le_radical J) (self_le_radical_diagram J) i).symm
+≪≫ has_colimit.iso_of_nat_iso (iso.refl _)
+
+/-- Casting from the full subcategory of ideals with radical containing `J` to the full
+subcategory of ideals with radical containing `K`. -/
+def self_le_radical.cast (hJK : J.radical = K.radical) :
+  self_le_radical J ⥤ self_le_radical K :=
+full_subcategory.map (λ L hL, begin
+                        rw ← ideal.radical_le_radical_iff at ⊢ hL,
+                        exact hJK.symm.trans_le hL,
+                      end)
+
+instance self_le_radical.cast_is_equivalence (hJK : J.radical = K.radical) :
+  is_equivalence (self_le_radical.cast hJK) :=
+{ inverse := self_le_radical.cast hJK.symm,
+  unit_iso := by tidy,
+  counit_iso := by tidy }
```

Comment:
I tried this and it got kind of long and none of the `obviously` proofs worked, so my approach seems flawed. It would probably be better to have this be a separate PR. Here's my attempt:
```lean
namespace category_theory

universes v u

variables {C : Type u} [category.{v} C]
variables {Z Z' Z'' : C → Prop}

/- TODO:  Once this file is ported, move this section to the right location. -/
def full_subcategory.iso_of_iso ⦃X Y : full_subcategory Z⦄ (f : X.obj ≅ Y.obj) : X ≅ Y :=
{ hom := f.hom, inv := f.inv, hom_inv_id' := f.hom_inv_id', inv_hom_id' := f.inv_hom_id' }

def full_subcategory.map_map' (h : ∀ ⦃X⦄, Z X → Z' X) (h' : ∀ ⦃X⦄, Z' X → Z'' X) :
  full_subcategory.map h ⋙ full_subcategory.map h' ≅ full_subcategory.map (λ X hX, h' (h hX)) :=
nat_iso.of_components
  (λ (X : full_subcategory Z), full_subcategory.iso_of_iso (iso.refl X.obj))
  (λ X Y f, (category.comp_id f).trans (category.id_comp f).symm)

def full_subcategory.map_refl (Z : C → Prop) :
  full_subcategory.map (λ ⦃X⦄, (id : Z X → Z X)) ≅ functor.id (full_subcategory Z) :=
nat_iso.of_components
  (λ X, full_subcategory.iso_of_iso (iso.refl X.obj))
  (λ X Y f, (category.comp_id f).trans (category.id_comp f).symm)

def full_subcategory.coe_map ⦃X Y : full_subcategory Z⦄ : (X ⟶ Y) → (X.obj ⟶ Y.obj) := id

def full_subcategory.map_eq_iff ⦃X Y : full_subcategory Z⦄ (f g : X ⟶ Y) :
f = g ↔ (full_subcategory.coe_map f) = (full_subcategory.coe_map g) := iff.rfl

@[simps]
def full_subcategory.equivalence (h : ∀ X, Z X ↔ Z' X) :
  full_subcategory Z ≌ full_subcategory Z' :=
{ functor := full_subcategory.map (λ X, (h X).mp),
  inverse := full_subcategory.map (λ X, (h X).mpr),
  unit_iso := (full_subcategory.map_map' (λ X, (h X).mp) (λ X, (h X).mpr)
                ≪≫ full_subcategory.map_refl Z).symm,
  counit_iso := (full_subcategory.map_map' (λ X, (h X).mpr) (λ X, (h X).mp)
                ≪≫ full_subcategory.map_refl Z'),
  functor_unit_iso_comp' := λ X, begin
    change (𝟙 X.obj ≫ 𝟙 X.obj) ≫ (𝟙 X.obj ≫ 𝟙 X.obj) = 𝟙 X.obj,
    simp
  end }
end category_theory
```

Reply:
Yes, this is fine, it can go in another PR. Could you just as a single line `TODO` at `self_le_radical.cast_is_equivalence`, something like ``-- TODO generalize this to the equivalence of full categories for any `iff`.``